indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java

```diff
@@ -139,6 +140,100 @@ public List<String> extractVocabLabelsFromThemes(List<ThemesModel> themes, Strin
         return results;
     }
 
+    public List<String> extractOrganisationVocabLabelsFromThemes(List<ThemesModel> themes) {
+        List<String> results = new ArrayList<>();
+        themes.stream().filter(Objects::nonNull).forEach(theme -> {
+            if (theme.getTitle().toLowerCase().contains("aodn organisation vocabulary")) {
+                for (ConceptModel conceptModel : theme.getConcepts()) {
+                    if (conceptModel.getId() != null && !conceptModel.getId().isEmpty()) {
+                        results.add(conceptModel.getId());
+                    }
+                }
+            }
+        });
+        return results;
+    }
+
+    public List<VocabModel> getMappedOrganisationVocabsFromContacts(List<ContactsModel> contacts) throws IOException {
+        List<String> contactOrgs = new ArrayList<>();
+        String citationRole = "citation";
+        String pointOfContactRole = "pointOfContact";
+
+        // top priority to citation: cit:citedResponsibleParty>
+        for (ContactsModel contact : contacts) {
+            if (contact.getRoles().contains(citationRole)) {
+                String contactOrg = contact.getOrganization();
+                if (contactOrg != null) {
+                    contactOrgs.add(contactOrg);
+                }
+            }
+        }
+
+        // second priority
+        if (contactOrgs.isEmpty()) {
+            for (ContactsModel contact : contacts) {
+                if (contact.getRoles().contains(pointOfContactRole)) {
+                    String contactOrg = contact.getOrganization();
+                    if (contactOrg != null) {
+                        contactOrgs.add(contactOrg);
+                    }
+                }
+            }
+        }
+
+        List<VocabModel> results = new ArrayList<>();
+        for (JsonNode orgVocab : self.getOrganisationVocabs()) {
+            if (orgVocab != null) {
+                try {
+                    VocabModel vocabModel = indexerObjectMapper.treeToValue(orgVocab, VocabModel.class);
+                    dfsSearch(vocabModel, contactOrgs, results);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException("Error deserializing JsonNode to VocabModel", e);
```

Comment:
Will crash the whole app with RuntimeException, can we handle it better or create the VocabModel on start so we do not need to process it each time and fail on start